WcaOnRails/app/controllers/competitions_controller.rb

```diff
@@ -345,19 +345,20 @@ def paypal_return
 
   def stripe_connect
     code = params[:code]
-    competition = Competition.find(params[:state])
-    unless current_user&.can_manage_competition?(competition)
+    @competition = Competition.find(params[:state])
```

Comment:
The `@` is only necessary if there is an accompanying Rails view that needs to access this variable.

In this case, the method always ends with a redirect, which leads to another controller method. A view is never shown directly from this controller method, so the `@` is unnecessary.

Reply:
My understanding is that `@` defines an instance variable - so I like using @ by default so that if we call another method inside the controller we can access the "sane" instance variables (stuff like @competition, @registration, etc). 

The thinking is that even if we're not _currently_ calling a method that needs access to an instance variable, I'd rather it be easy to implement in future.

But if you'd rather we explicitly pass stuff as parameters only as the need arises, I'm happy to do that!

Reply:
Your technical explanations are all correct. The Rails templates/views can see it _because_ it is an instance variable of the controller.

> But if you'd rather we explicitly pass stuff as parameters only as the need arises, I'm happy to do that!

Yes please! The argument that "somebody might need this at some point in the future" leads to pollution and cluttering of source code. If you follow this paradigm all the time, you will end up with a scenario where suddenly there's a superfuous instance variable and you don't know where it was defined.

Reply:
Makes sense, will do!